src/foundation.cs

```diff
@@ -16086,7 +16086,155 @@ interface NSMeasurementFormatter : NSSecureCoding {
+
+		[Export ("initWithListenerEndpoint:")]
+		IntPtr Constructor (NSXpcListenerEndpoint endpoint);
+
+		[Export ("endpoint")]
+		NSXpcListenerEndpoint Endpoint { get; }
+
+		[Export ("exportedInterface", ArgumentSemantic.Retain)]
+		[NullAllowed]
+		NSXpcInterface ExportedInterface { get; set; }
+
+		[Export ("exportedObject", ArgumentSemantic.Retain)]
+		[NullAllowed]
+		NSObject ExportedObject { get; set; }
+
+		[Export ("remoteObjectInterface", ArgumentSemantic.Retain)]
+		[NullAllowed]
+		NSXpcInterface RemoteInterface { get; set; }
+
+		[Export ("interruptionHandler", ArgumentSemantic.Copy)]
+		Action InterruptionHandler { get; set; }
+
+		[Export ("invalidationHandler", ArgumentSemantic.Copy)]
+		Action InvalidationHandler { get; set; }
+
+		[Export ("resume")]
+		void Resume ();
+
+		[Export ("suspend")]
+		void Suspend ();
+
+		[Export ("invalidate")]
+		void Invalidate ();
+
+		[Export ("auditSessionIdentifier")]
+		int AuditSessionIdentifier { get; }
+
+		[Export ("processIdentifier")]
+		int PeerProcessIdentifier { get; }
+
+		[Export ("effectiveUserIdentifier")]
+		int PeerEffectiveUserId { get; }
+
+		[Export ("effectiveGroupIdentifier")]
+		int PeerEffectiveGroupId { get; }
+
+		[Export ("currentConnection")]
+		[Static]
+		[Mac (10, 8)][iOS (6, 0)][Watch (2, 0)][TV (9, 0)]
+		NSXpcConnection CurrentConnection { [return: NullAllowed] get; }
+
+		[Export ("scheduleSendBarrierBlock:")]
+		[Mac (10, 15)][iOS (13, 0)][Watch (6, 0)][TV (13, 0)]
+		void ScheduleSendBarrier (Action block);
+
+		[Export ("remoteObjectProxy"), Internal]
+		IntPtr _CreateRemoteObjectProxy ();
+
+		[Export ("remoteObjectProxyWithErrorHandler:"), Internal]
+		IntPtr _CreateRemoteObjectProxy ([BlockCallback] Action<NSError> errorHandler);
+
+		[Mac (10, 11)][iOS (9, 0)][Watch (2, 0)][TV (9, 0)]
+		[Export ("synchronousRemoteObjectProxyWithErrorHandler:"), Internal]
+		IntPtr _CreateSynchronousRemoteObjectProxy ([BlockCallback] Action<NSError> errorHandler);
+	}
+
+	[Mac (10, 8)][iOS (6, 0)][Watch (2, 0)][TV (9, 0)]
+	[BaseType (typeof (NSObject), Name = "NSXPCListener", Delegates = new string[] { "WeakDelegate" })]
+	[DisableDefaultCtor]
+	interface NSXpcListener
+	{
+		[Export ("serviceListener")]
+		[Static]
+		NSXpcListener ServiceListener { get; }
+
+		[Export ("anonymousListener")]
+		[Static]
+		NSXpcListener AnonymousListener { get; }
+
+		[Export ("initWithMachServiceName:")]
+		[NoiOS][NoTV][NoWatch]
+		IntPtr Constructor (string machServiceName);
+
+		[Export ("delegate", ArgumentSemantic.Assign)]
+		[NullAllowed]
+		NSObject WeakDelegate { get; set; }
+
+		[Wrap ("WeakDelegate")]
+		[Protocolize]
+		NSXpcListenerDelegate Delegate { get; set; }
+
+		[Export ("endpoint")]
+		NSXpcListenerEndpoint Endpoint { get; }
+
+		[Export ("resume")]
+		void Resume ();
+
+		[Export ("suspend")]
+		void Suspend ();
+
+		[Export ("invalidate")]
+		void Invalidate ();
+	}
+
+	[BaseType (typeof (NSObject), Name = "NSXPCListenerDelegate")]
+	[Model, Protocol]
```

Comment:
This is just a technically we have to do:

```suggestion
	[Model (AutoGeneratedName = true), Protocol]
```